### PR TITLE
tests: add tests for mail, url, and rule validation utilities

### PR DESCRIPTION
**Pull request by elie222**
# User description
## Summary

Add 83 new tests covering critical utility functions that previously lacked test coverage.

- **mail.test.ts** (26 tests): Email parsing, content extraction, HTML-to-text conversion
- **url.test.ts** (22 tests): Provider-specific URL building for Gmail/Outlook
- **rule.validation.test.ts** (35 tests): Zod schema validation for rule creation

## Bug Fixes

- Fix whitespace-only name validation in rule schema (added `.trim()`)
- Remove ineffective `.default()` from `conditionalOperator` field

## Test plan

- [x] All 1094 tests pass
- [x] New tests cover edge cases and document existing behavior

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
Adds comprehensive test coverage for core utility functions, including email parsing, URL generation, and rule validation schemas. Refines rule validation logic and action sorting behavior.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1235?tool=ast&topic=Rule+Validation>Rule Validation</a>
        </td><td>Adds extensive tests for the <code>createRuleBody</code> and <code>delayInMinutesSchema</code> Zod schemas, ensuring robust validation for rule creation data. Fixes a bug in <code>createRuleBody</code> where rule names with only whitespace were considered valid by adding <code>.trim()</code>, and removes an ineffective <code>.default()</code> value from the <code>conditionalOperator</code> field.<details><summary>Modified files (2)</summary><ul><li>apps/web/utils/actions/rule.validation.test.ts</li>
<li>apps/web/utils/actions/rule.validation.ts</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>Add-cold-email-notifier</td><td>December 19, 2025</td></tr>
<tr><td>joshwerner001@gmail.com</td><td>Cleaning-up-the-user-e...</td><td>December 01, 2025</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1235?tool=ast&topic=General+Utilities>General Utilities</a>
        </td><td>Adds new tests for various other utility functions, including action item field handling (<code>getActionFields</code>, <code>sanitizeActionFields</code>), action sorting logic (<code>sortActionsByPriority</code>), ignored sender filtering (<code>isIgnoredSender</code>), CTA and unsubscribe keyword detection (<code>containsCtaKeyword</code>, <code>containsUnsubscribeKeyword</code>), template variable parsing (<code>hasVariables</code>), and text slugification (<code>slugify</code>). Also updates the <code>ACTION_PRIORITY_ORDER</code> to include <code>ActionType.NOTIFY_SENDER</code> and modifies <code>containsUnsubscribeKeyword</code> for case-insensitive matching.<details><summary>Modified files (9)</summary><ul><li>apps/web/utils/action-item.test.ts</li>
<li>apps/web/utils/action-sort.test.ts</li>
<li>apps/web/utils/action-sort.ts</li>
<li>apps/web/utils/filter-ignored-senders.test.ts</li>
<li>apps/web/utils/parse/cta.test.ts</li>
<li>apps/web/utils/parse/unsubscribe.test.ts</li>
<li>apps/web/utils/parse/unsubscribe.ts</li>
<li>apps/web/utils/template.test.ts</li>
<li>apps/web/utils/text.test.ts</li></ul></details><details><summary>Latest Contributors(1)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>fix-up-imports-for</td><td>November 22, 2025</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1235?tool=ast&topic=Email+%26+URL+Utilities>Email & URL Utilities</a>
        </td><td>Introduces new tests for email-related utilities, covering content extraction (<code>emailToContent</code>), HTML-to-text conversion (<code>convertEmailHtmlToText</code>), reply parsing (<code>parseReply</code>), and email client identification (<code>getEmailClient</code>). Also adds tests for URL generation functions, ensuring correct construction of Gmail and Outlook links, search queries, and filter settings URLs.<details><summary>Modified files (2)</summary><ul><li>apps/web/utils/mail.test.ts</li>
<li>apps/web/utils/url.test.ts</li></ul></details><details><summary>Latest Contributors(0)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1235?tool=ast>(Baz)</a>.

## Summary by CodeRabbit

* **Bug Fixes**
  * Unsubscribe keyword detection now works case-insensitively for improved matching
  * Rule name validation now properly trims leading and trailing whitespace

* **New Features**
  * Added NOTIFY_SENDER action type to action priority ordering

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/web/utils/mail.test.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  it("extracts visible text from email reply", () => {
+    const plainText = `New message here
+
+On Jan 1, 2024, someone@example.com wrote:
+> Old quoted content
+> More quoted stuff`;
+
+    const result = parseReply(plainText);
+    expect(result).toContain("New message here");
+    // The library should strip quoted content
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/mail.test.ts]: Good catch! Added assertions to verify the quoted content is actually stripped:
```typescript
expect(result).not.toContain("Old quoted content");
expect(result).not.toContain("More quoted stuff");
```
